pkg/extensions/sync/sync.go

```diff
@@ -0,0 +1,363 @@
+package sync
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/anuvu/zot/errors"
+	"github.com/anuvu/zot/pkg/log"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/types"
+	"gopkg.in/resty.v1"
+)
+
+var postChan chan bool
+
+// /v2/_catalog struct.
+type catalog struct {
+	Repositories []string `json:"repositories"`
+}
+
+type Config []RegistryConfig
+
+type RegistryConfig struct {
+	URL          string
+	PollInterval time.Duration
+	Content      []Content
+}
+
+type Content struct {
+	Prefix string
+	Tags   *Tags
+}
+
+type Tags struct {
+	Regex  *string
+	Semver *bool
+}
+
+func Handler(w http.ResponseWriter, r *http.Request) {
+	postChan <- true
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write([]byte{})
+}
+
+// getCatalog gets all repos from a registry.
+func getCatalog(registryName string) (catalog, error) {
+	var c catalog
+
+	registryCatalogURL := fmt.Sprintf("%s%s%s", "https://", registryName, "/v2/_catalog")
+	client := resty.New()
```

Comment:
we need proxy support else this whole sync feature is un-usable.

Reply:
I've tested it, and it reads the proxy config from env variables. Do you think we still need the config?